python/GafferUI/SpreadsheetUI.py

```diff
@@ -1215,6 +1319,307 @@ def __fixedWidth( cls, plugValueWidget ) :
 
 		return None
 
+# _SectionChooser
+# ===============
+
+class _SectionChooser( GafferUI.Widget ) :
+
+	def __init__( self, rowsPlug, **kw ) :
+
+		tabBar = QtWidgets.QTabBar()
+		GafferUI.Widget.__init__( self, tabBar, **kw )
+
+		self.__rowsPlug = rowsPlug
+		self.__currentSectionChangedSignal = Gaffer.Signal1()
+
+		tabBar.setDrawBase( False )
+		tabBar.setMovable( True )
+		tabBar.setExpanding( False )
+		tabBar.setUsesScrollButtons( False )
+
+		tabBar.setContextMenuPolicy( QtCore.Qt.CustomContextMenu )
+		tabBar.customContextMenuRequested.connect( Gaffer.WeakMethod( self.__contextMenuRequested ) )
+
+		tabBar.currentChanged.connect( Gaffer.WeakMethod( self.__currentChanged ) )
+		self.__ignoreCurrentChanged = False
+		tabBar.tabMoved.connect( Gaffer.WeakMethod( self.__tabMoved ) )
+		self.__plugMetadataChangedConnection = Gaffer.Metadata.plugValueChangedSignal().connect( Gaffer.WeakMethod( self.__plugMetadataChanged ) )
+		self.__rowsPlug["default"]["cells"].childAddedSignal().connect( Gaffer.WeakMethod( self.__columnAdded ), scoped = False )
+		self.__rowsPlug["default"]["cells"].childRemovedSignal().connect( Gaffer.WeakMethod( self.__columnRemoved ), scoped = False )
+
+		self.__updateTabs()
+
+	def currentSection( self ) :
+
+		if not self._qtWidget().count() :
+			return None
+
+		return self._qtWidget().tabText( self._qtWidget().currentIndex() )
+
+	def currentSectionChangedSignal( self ) :
+
+		return self.__currentSectionChangedSignal
+
+	@classmethod
+	def sectionNames( cls, rowsPlug ) :
+
+		names = set()
+		for cellPlug in rowsPlug["default"]["cells"] :
+			names.add( cls.getSection( cellPlug ) )
+
+		if names == { "Other" } :
+			# "Other" is a special section that we put columns into
+			# automatically if they don't have any section metadata.
+			# This helps us be robust to columns that are added directly
+			# through the API by code unaware of sectioning. When _everything_
+			# is in "Other", we automatically disable sectioning.
+			return []
+
+		namesAndIndices = []
+		for name in sorted( list( names ) ) :
+			index = len( namesAndIndices ) if name != "Other" else sys.maxsize
```

Comment:
Is this attempting to keep `Other` at the end? Its not critical, but a second section added to a sheet will appear after `Other` in the bar due to the implementation of `__assignSectionOrder` and how its called in `setSection`. If we skipped the next line if `name == "Other"` I think we'd be there. Its not a blocker for release though, just a nicety.

Reply:
It's putting `Other` at the end when `Other` has been created implicitly (through the addition of columns via the API rather than UI). We can't skip the next line because then the ordering will be ignored if the user specifies the position of `Other` explicitly (by dragging). But I've pushed 1d585c1 to adjust the ordering logic in `setSection()` so that if `Other` was last before a section is added, it stays last. This also fixes another little buglet which meant we didn't clean up the index metadata for old sections when a new section was being created.